osquery/core/tables.cpp

```diff
@@ -395,48 +410,38 @@ std::set<std::string> ConstraintList::getAll(ConstraintOperator op) const {
   return set;
 }
 
-template <>
-std::set<int> ConstraintList::getAll<int>(ConstraintOperator op) const {
-  std::set<int> cs;
-  std::transform(constraints_.begin(),
-                 constraints_.end(),
-                 std::inserter(cs, cs.begin()),
-                 [](const Constraint& c) { return std::stoi(c.expr); });
+template <typename T>
+std::set<T> ConstraintList::getAll(ConstraintOperator op) const {
+  std::set<T> cs;
+  auto mc_ = constraints_;
+  std::transform(
+      mc_.begin(),
+      std::remove_if(mc_.begin(),
+                     mc_.end(),
+                     [](const Constraint& c) { return !tryTo<T>(c.expr); }),
+      std::inserter(cs, cs.begin()),
+      [](const Constraint& c) { return tryTo<T>(c.expr).take(); });
```

Comment:
I think it could be written much simpler:
```c++
for (const auto& item : constraints_) {
  auto exp = tryTo<T>(item.expr);
  if (exp) {
    cs.insert(exp.take());
  }
}
```
What do you think?